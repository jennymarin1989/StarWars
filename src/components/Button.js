import React from 'react';
import StyleButton from './styleComponents/StyleButton';
import './styleComponents/Button.css'


const Button = (props) => {

  return (

    <div>
      <StyleButton>
      <h2 className="button-header">CLick here to search for People!</h2>
        <button 
          id='homepage-button' 
          onClick={props.onClick}>
            <img src={props.image} className="button-image" alt="starwars"/>
        </button>
      </StyleButton >
    </div>  
  )
}
export default Button;