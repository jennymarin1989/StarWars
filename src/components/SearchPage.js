import React, { Component } from 'react';

class SearchPage extends Component {

  render () {
    return (
      <div>
        <h2>Starwars search</h2>
      </div>  
    );
  }
}

export default SearchPage;