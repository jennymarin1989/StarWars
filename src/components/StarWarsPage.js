import React, { Component } from 'react';

class StarWarsPage extends Component {
  

  render () {
    return (
      <div>
      </div>  
    );
  }
}

export default StarWarsPage;