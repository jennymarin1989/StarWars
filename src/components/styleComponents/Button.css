.button-header {
  color: #FFD700
}

.button-image {
  width: 13%;
  height: 20%
}