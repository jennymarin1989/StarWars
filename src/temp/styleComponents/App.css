.grid-style {
  background-size: cover;
  background-position: center;
  width: 100%;
  height: -webkit-fill-available; 
}

.row-title {
  background-color: white;
  padding-bottom: 3%
}

.row-button {
  margin-top: 5%
}

.stars-page  {
  margin-top: 2.5%
}