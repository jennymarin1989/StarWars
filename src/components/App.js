import React, { Component } from 'react';
import Title from '../components/Title';
import Button from '../components/Button';
import StarWarsPage from '../components/StarWarsPage';
import {Grid, Row}  from 'react-bootstrap';
import Image from '../dark.jpg';
import './styleComponents/App.css';

class App extends Component {
  constructor(){
    super();
    this.state = {
      starwars: false
    }
   
    this.handleClick= this.handleClick.bind(this) 
  }
  
  handleClick() {
    this.setState ({
      starwars: !false
    })
  }
 
  render () {
    return (
      <Grid 
        width={500}
        align="center"
        className="grid-style"
        style={{backgroundImage: `url(${Image})`}} >
          <Row width={100} className = "row-title">
            <Title header = {"LEGO"}/>
          </Row> 
          <Row width={100} className="row-button"> 
            <Button 
              image={"https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Star_Wars_Logo.svg/1200px-Star_Wars_Logo.svg.png"}
              onClick={this.handleClick}
            />
          </Row>  
          <Row width={120} className="stars-page">  
            {this.state.starwars ? <StarWarsPage /> : null}
          </Row> 
      </Grid>
    )
  }
  
}

export default App;