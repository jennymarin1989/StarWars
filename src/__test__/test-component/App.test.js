import React from 'react';
import { shallow } from 'enzyme';
import App from '../../components/App';
import Button from '../../components/Button';
import SearchPage from '../../components/StarWarsPage';
import StarWarsPage from '../../components/StarWarsPage';

describe('App', () => {

  let app;

    beforeEach(() => {
      app = shallow(<App />);
    })

  describe('renders correctly', () => {

    it('renders components correctly', () => {
    expect(app).toMatchSnapshot();  
    });
  
    it('renders Title component correctly', () => {
     expect(app.find('Title')).toHaveLength(1);
    });

    it('renders Button component correctly', () => {
      expect(app.find('Button')).toHaveLength(1);
    });
  });

  describe('Initializes state correctly', () => {
    it('initializes state of starwars button', () => {
      expect(app.state('starwars')).toBeFalsy();
    });
  });

  describe('changes state correctly', () => {
    it('changes starwars state on click', () => {
      const mockCLickButton = jest.fn();
      let button2 = shallow(<Button onclick={mockCLickButton} />);
      button2.find('#homepage-button').simulate('click');
      expect(mockCLickButton.mock.calls.length).toEqual(1)
    })
  });

  describe ('renders searchpage', () => {
    it ('renders searchpage if starwars state is true', () => {
      app.setState({ starwars: true });
      expect(app.find(StarWarsPage).exists()).toBe(true);
    })
  });

})