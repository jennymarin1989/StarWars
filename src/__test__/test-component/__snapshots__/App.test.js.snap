// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly renders components correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "align": "center",
      "bsClass": "container",
      "children": Array [
        <Row
          bsClass="row"
          className="row-title"
          componentClass="div"
          width={100}
>
          <Title
                    header="LEGO"
          />
</Row>,
        <Row
          bsClass="row"
          className="row-button"
          componentClass="div"
          width={100}
>
          <Button
                    image="https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Star_Wars_Logo.svg/1200px-Star_Wars_Logo.svg.png"
                    onClick={[Function]}
          />
</Row>,
        <Row
          bsClass="row"
          className="stars-page"
          componentClass="div"
          width={120}
/>,
      ],
      "className": "grid-style",
      "componentClass": "div",
      "fluid": false,
      "style": Object {
        "backgroundImage": "url(dark.jpg)",
      },
      "width": 500,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <Title
            header="LEGO"
/>,
          "className": "row-title",
          "componentClass": "div",
          "width": 100,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "header": "LEGO",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <Button
            image="https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Star_Wars_Logo.svg/1200px-Star_Wars_Logo.svg.png"
            onClick={[Function]}
/>,
          "className": "row-button",
          "componentClass": "div",
          "width": 100,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "image": "https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Star_Wars_Logo.svg/1200px-Star_Wars_Logo.svg.png",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": null,
          "className": "stars-page",
          "componentClass": "div",
          "width": 120,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "center",
        "bsClass": "container",
        "children": Array [
          <Row
            bsClass="row"
            className="row-title"
            componentClass="div"
            width={100}
>
            <Title
                        header="LEGO"
            />
</Row>,
          <Row
            bsClass="row"
            className="row-button"
            componentClass="div"
            width={100}
>
            <Button
                        image="https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Star_Wars_Logo.svg/1200px-Star_Wars_Logo.svg.png"
                        onClick={[Function]}
            />
</Row>,
          <Row
            bsClass="row"
            className="stars-page"
            componentClass="div"
            width={120}
/>,
        ],
        "className": "grid-style",
        "componentClass": "div",
        "fluid": false,
        "style": Object {
          "backgroundImage": "url(dark.jpg)",
        },
        "width": 500,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": <Title
              header="LEGO"
/>,
            "className": "row-title",
            "componentClass": "div",
            "width": 100,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "header": "LEGO",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": <Button
              image="https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Star_Wars_Logo.svg/1200px-Star_Wars_Logo.svg.png"
              onClick={[Function]}
/>,
            "className": "row-button",
            "componentClass": "div",
            "width": 100,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "image": "https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Star_Wars_Logo.svg/1200px-Star_Wars_Logo.svg.png",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": null,
            "className": "stars-page",
            "componentClass": "div",
            "width": 120,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
